src/python/pants/jvm/target_types.py

```diff
@@ -123,6 +128,20 @@ class JvmArtifact(Target):
     )
 
 
+# -----------------------------------------------------------------------------------------------
+# JUnit test support field(s)
+# -----------------------------------------------------------------------------------------------
+
+
+class JunitTestSourceField(SingleSourceField, metaclass=ABCMeta):
```

Comment:
It's confusing imo for this to subclass `SingleSourceField` and then for the subclasses of this to also subclass `JavaSourceField` and `ScalaSourceField`. it works, but confusing.

Instead, I recommend calling this `JunitTestSourceMixin`, and it's nothing more than a marker type. Doesn't subclass anything, not even `ABCMeta`.

Reply:
That won't work I don't think: this does actually need to be a `SourceField` subclass, because it is used with a field set.